File: unity/oelMVCS/Assets/XTC/oelMVCS/Scripts/View/ViewCenter.cs
using System.Collections.Generic;

namespace XTC.oelMVCS
{
    public class ViewCenter
    {
        private Board board_
        {
            get;
            set;
        }

        private Dictionary<string, View.Inner> views = new Dictionary<string, View.Inner>();

        public ViewCenter(Board _board)
        {
            board_ = _board;
        }

        public Error Register(string _uuid, View.Inner _view)
        {
            board_.logger.Info("register view {0}", _uuid);
            if (views.ContainsKey(_uuid))
                return Error.NewAccessErr("view {0} exists", _uuid);
            views[_uuid] = _view;
            return Error.OK;
        }

        public Error Cancel(string _uuid)
        {
            board_.logger.Info("cancel view {0}", _uuid);
            if (!views.ContainsKey(_uuid))
                return Error.NewAccessErr("view {0} not found", _uuid);
            views.Remove(_uuid);
            return Error.OK;
        }


        public View.Inner FindView(string _uuid)
        {
            if (views.ContainsKey(_uuid))
                return views[_uuid];
            return null;
        }


        public void Setup()
        {
            foreach (View.Inner inner in views.Values)
            {
                inner.Setup(board_);
            }
        }

        public void Dismantle()
        {
            foreach (View.Inner view in views.Values)
            {
                view.Dismantle();
            }
        }

        public void HandleAction(string _action, Model.Status _status, object _obj)
        {
            foreach (View.Inner inner in views.Values)
            {
                inner.Handle(_action, _status, _obj);
            }
        }
    }
}//namespace

File: python/oelMVCS/ViewCenter.py
# -*- coding: utf-8 -*-

class ViewCenter:

    def __init__(self, _framework):
         self.__framework = _framework
         self.views = {}

    def Register(self, _uuid, _view):
        self.__framework.logger.Info('register view %s'%(_uuid))
        if _uuid in self.views:
            raise KeyError('view is exists')
        self.views[_uuid] = _view

    def Cancel(self, _uuid):
        self.__framework.logger.Info('cancel view %s'%(_uuid))
        if not _uuid in self.views:
            raise KeyError('view not found')
        del self.views[_uuid] 

    def Find(self, _uuid):
        return self.views[_uuid]

    def Setup(self):
        for v in self.views.values():
            v.Setup(self.__framework)

    def Dismantle(self):
        for v in self.views.values():
            v.Dismantle()

    def handleAction(self, _action, _status, _data):
        for v in self.views.values():
            v.handle(_action, _status, _data)


